### PR TITLE
Regex refactor

### PR DESCRIPTION
**Pull request by hbmartin**
* Remove `groups` and `flags` parameters from `regex_search` function.
* No branching return types from `regex_search`
* Directly implement regex's for the 3 places that used the above parameters
* Added tests

### CODE REVIEW

**Comment by hbmartin** [comment]
@movraxrsp @swiftyy-mage @DmitryBurnaev can you take a look at this PR?

File: pytube/cipher.py [hunk 1]
```diff
@@ -262,17 +279,16 @@ def get_signature(js: str, ciphered_signature: str) -> str:
...
+        logger.debug("applied transform function\n")
+        # pprint.pformat(
+        #     {
+        #         "output": "".join(signature),
+        #         "js_function": name,
+        #         "argument": int(argument),
+        #         "function": transform_map[name],
+        #     },
+        #     indent=2,
+        # ),
```
Comment (DmitryBurnaev) [inline pytube/cipher.py]: Could you remove this debug instruction?

File: pytube/cipher.py [hunk 2]
```diff
@@ -104,14 +110,17 @@ def get_transform_object(js, var):
...
-        regex_search(pattern, js, group=1, flags=re.DOTALL)
-        .replace("\n", " ")
-        .split(", ")
-    )
+    regex = re.compile(pattern, flags=re.DOTALL)
+    results = regex.search(js)
+    if not results:
+        raise RegexMatchError(
+            "regex pattern ({pattern}) had zero matches".format(pattern=pattern),
+        )
```
Comment (DmitryBurnaev) [inline pytube/cipher.py]: May be we can use parametrized constructor for `RegexMatchError`? 
like...
```
...
raise RegexMatchError(pattern)
```

File: pytube/helpers.py [hunk 3]
```diff
@@ -7,69 +7,36 @@
...
     :param int group:
         Index of group to return.
-    :param int flags:
-        Expression behavior modifiers.
     :rtype:
         str or tuple
     :returns:
         Substring pattern matches.
     """
-    if type(pattern) == list:
```
Comment (DmitryBurnaev) [inline pytube/helpers.py]: multiple patterns won't be available, right?

Comment (hbmartin) [reply pytube/helpers.py]: @DmitryBurnaev nope, the list was only used in cipher.get_initial_function_name, which now uses `re` directly

File: pytube/cipher.py (hunk 2 above)
Comment (hbmartin) [reply pytube/cipher.py]: @DmitryBurnaev good idea, rewrote this and other exceptions in e7518ec58dc8eef8a6653af696d2c57eab770db8

File: pytube/cipher.py (hunk 1 above)
Comment (hbmartin) [reply pytube/cipher.py]: @DmitryBurnaev derp, removed commented code and logging parameters in 868b06442082cc02333ff0627ce7d322987557d3

**Comment by hbmartin** [comment]
Merging now, happy to fix any additional comments.

**Comment by movraxrsp** [comment]
I just finished checking it all out after my workday.

I like it. Much better than what's running on pytube now. Nice work.

**Pull request merged**